pkg/controllers/machine/disruption/drift.go

```diff
@@ -100,3 +103,43 @@ func areStaticFieldsDrifted(provisioner *v1alpha5.Provisioner, machine *v1alpha5
 	}
 	return provisionerHash != machineHash
 }
+
+func areNodeRequirementsDrifted(provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) bool {
+	return !lo.EveryBy(provisioner.Spec.Requirements, func(provisionerReq v1.NodeSelectorRequirement) bool {
+		value, found := machine.Labels[provisionerReq.Key]
+		if found {
+			if provisionerReq.Operator == v1.NodeSelectorOpGt {
+				return driftedNRGreaterThen(provisionerReq.Values[0], value)
+			}
+			if provisionerReq.Operator == v1.NodeSelectorOpLt {
+				return driftedNRLessThen(provisionerReq.Values[0], value)
+			}
+			if provisionerReq.Operator == v1.NodeSelectorOpIn {
+				return lo.Contains(provisionerReq.Values, value)
+			}
+			if provisionerReq.Operator == v1.NodeSelectorOpNotIn {
+				return !lo.Contains(provisionerReq.Values, value)
+			}
+			return provisionerReq.Operator == v1.NodeSelectorOpExists
+		}
+		return provisionerReq.Operator == v1.NodeSelectorOpDoesNotExist
+	})
+}
+
+func driftedNRGreaterThen(provisionerValue string, machineValue string) bool {
+	valueInt, errMach := strconv.Atoi(machineValue)
+	provValueInt, errProv := strconv.Atoi(provisionerValue)
+	if errMach != nil && errProv != nil {
+		return false
+	}
+	return provValueInt < valueInt
+}
+
+func driftedNRLessThen(provisionerValue string, machineValue string) bool {
+	valueInt, errMach := strconv.Atoi(machineValue)
+	provValueInt, errProv := strconv.Atoi(provisionerValue)
+	if errMach != nil && errProv != nil {
+		return false
+	}
+	return provValueInt > valueInt
+}
```

Comment:
These are essentially the same function, consider parameterizing the difference into one function:
```suggestion
func nodeRequirementsCompare(provisionerValue string, machineValue string, lt bool) bool {
	valueInt, errMach := strconv.Atoi(machineValue)
	provValueInt, errProv := strconv.Atoi(provisionerValue)
	if errMach != nil && errProv != nil {
		return false
	}
	if lt {
	    return provValueInt > valueInt
	}
	return provValueInt < valueInt
}
```